include/arc_utilities/ros_helpers.hpp

```diff
@@ -115,6 +132,40 @@ namespace ROSHelpers
             return Maybe::Maybe<T>();
         }
     }
+
+    template <typename T>
+    inline Maybe::Maybe<std::vector<T>> GetVectorOptional(const ros::NodeHandle& nh, const std::string& param_name, const std::string& calling_fn_name)
+    {
+        ROS_DEBUG_STREAM_NAMED("params", "No default value for " << param_name << ": Value must be on paramter sever");
+        std::vector<T> vec;
+        if (nh.getParam(param_name, vec))
+        {
+            ROS_INFO_STREAM_NAMED("params", "Retrieving " << std::left << std::setw(PARAM_NAME_WIDTH) << param_name << " of length " << vec.size());
+            return Maybe::Maybe<std::vector<T>>(vec);
+        }
+        else
+        {
+            ROS_INFO_STREAM_NAMED("params", "Cannot find " << nh.getNamespace() << "/" << param_name << " on parameter server for " << calling_fn_name << ": Value must be on paramter sever");
+            return Maybe::Maybe<std::vector<T>>();
+        }
+    }
+
+    template <typename T>
+    inline Maybe::Maybe<std::vector<T>> GetVectorRequired(const ros::NodeHandle& nh, const std::string& param_name, const std::string& calling_fn_name)
```

Comment:
With a name like `GetVectorRequired` I'd expect a more extreme handling than just "writes to ROS_FATAL rather than ROS_INFO"

With this name I'd expect the interface to be:

```
// throws error if vector not on parameter server
std::vector<T> GetVectorRequired(...)
```

Reply:
Agreed. I'll file a new PR with changes to these new functions and the old ones.